src/sql/azdata.proposed.d.ts

```diff
@@ -432,6 +432,21 @@ declare module 'azdata' {
 		azurePortalEndpoint?: string;
 	}
 
+
+	export namespace diagnostics {
+		/**
+		 * Diagnostics object for handling error codes for a provider.
+		 */
+		export interface ErrorDiagnostics {
+			handleErrorCode(errorCode: number, errorMessage: string, additionalObjects?: any): Thenable<boolean>;
```

Comment:
This should have doc comments as well.

And is there any reason you don't have an actual profile parameter? This should always have a profile (because it's a connection error after all) so it seems odd to have this be "any" and then leave it up to the caller to guess/hope what the actual type is. 

What other "objects" are you expecting to pass in here?

Reply:
Can we rename this to `handleError` instead? Since its handling more than error code..

Reply:
@charles-gagnon I was expecting this service to be used for ANY error in ADS, not just for connection, but if it's solely for connection errors, then I guess I can change it to connection profile if needed.